org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -131,6 +134,17 @@ public void disconnectClient() {
 		return CompletableFuture.completedFuture(handler.initialize(params));
 	}
 
+	/*
+	 * (non-Javadoc)
+	 * @see org.eclipse.lsp4j.services.LanguageServer#initialized(org.eclipse.lsp4j.InitializedParams)
+	 */
+	@Override
+	public void initialized(InitializedParams params) {
```

Comment:
why registering the capability here instead of being consistent and do it with the other ones during initialization?

Reply:
It's not one of the capabilities that you can enable/disable on initialization.
I also don't like that, but that's how it currently is in the LSP.